src/test/unit/cli_unittest.cc

```diff
@@ -205,6 +222,283 @@ TEST(CLIUnittest, TestCliSetStringWriteOnce)
+        "aux 1 0 0 900 900 0 0", "\r\n",
+        "aux 2 0 0 900 900 0 0", "\r\n",
+        "aux 3 0 0 900 900 0 0", "\r\n",
+        "aux 4 0 0 900 900 0 0", "\r\n",
+        "aux 5 0 0 900 900 0 0", "\r\n",
+        "aux 6 0 0 900 900 0 0", "\r\n",
+        "aux 7 0 0 900 900 0 0", "\r\n",
+        "aux 8 0 0 900 900 0 0", "\r\n",
+        "aux 9 0 0 900 900 0 0", "\r\n",
+        "aux 10 0 0 900 900 0 0", "\r\n",
+        "aux 11 0 0 900 900 0 0", "\r\n",
+        "aux 12 0 0 900 900 0 0", "\r\n",
+        "aux 13 0 0 900 900 0 0", "\r\n",
+        "aux 14 0 0 900 900 0 0", "\r\n",
+        "aux 15 0 0 900 900 0 0", "\r\n",
+        "aux 16 0 0 900 900 0 0", "\r\n",
+        "aux 17 0 0 900 900 0 0", "\r\n",
+        "aux 18 0 0 900 900 0 0", "\r\n",
+        "aux 19 0 0 900 900 0 0", "\r\n",
+    };
+    EXPECT_EQ(expected, outLines);
+}
+
+TEST_F(AuxCliWriteTest, PrintAux_DiffAll)
+{
+    modeActivationCondition_t modifiedConditions[MAX_MODE_ACTIVATION_CONDITION_COUNT] = {
+        [0] = {
+            .modeId = BOXANGLE,
+            .auxChannelIndex = 7,
+            .range = { .startStep = 0, .endStep = 12 },
+            .modeLogic = MODELOGIC_OR,
+            .linkedTo = BOXARM
+        },
+    };
+    const char heading[] = "aux";
+    printAux(DO_DIFF, modifiedConditions, modeActivationConditions(0), heading);
+    vector<string> expected = {
+        "\r\n# ", "aux", "\r\n",
+        "aux 0 1 7 900 1200 0 0", "\r\n",
+    };
+    EXPECT_EQ(expected, outLines);
+}
+
+TEST_F(AuxCliWriteTest, Show)
+{
+    const char cmd[] = "aux";
+    char args[] = "";
+    cliAux(cmd, args);
+    vector<string> expected = {
+        "aux 0 0 0 900 900 0 0", "\r\n",
+        "aux 1 0 0 900 900 0 0", "\r\n",
+        "aux 2 0 0 900 900 0 0", "\r\n",
+        "aux 3 0 0 900 900 0 0", "\r\n",
+        "aux 4 0 0 900 900 0 0", "\r\n",
+        "aux 5 0 0 900 900 0 0", "\r\n",
+        "aux 6 0 0 900 900 0 0", "\r\n",
+        "aux 7 0 0 900 900 0 0", "\r\n",
+        "aux 8 0 0 900 900 0 0", "\r\n",
+        "aux 9 0 0 900 900 0 0", "\r\n",
+        "aux 10 0 0 900 900 0 0", "\r\n",
+        "aux 11 0 0 900 900 0 0", "\r\n",
+        "aux 12 0 0 900 900 0 0", "\r\n",
+        "aux 13 0 0 900 900 0 0", "\r\n",
+        "aux 14 0 0 900 900 0 0", "\r\n",
+        "aux 15 0 0 900 900 0 0", "\r\n",
+        "aux 16 0 0 900 900 0 0", "\r\n",
+        "aux 17 0 0 900 900 0 0", "\r\n",
+        "aux 18 0 0 900 900 0 0", "\r\n",
+        "aux 19 0 0 900 900 0 0", "\r\n",
+    };
+    EXPECT_EQ(expected, outLines);
+
+    for (int i = 0; i < MAX_MODE_ACTIVATION_CONDITION_COUNT; i++) {
+        auto condition = modeActivationConditions(i);
+        EXPECT_EQ(0, condition->auxChannelIndex);
+    }
+}
+
+TEST_F(AuxCliWriteTest, NotEnoughArgs)
+{
+    const char cmd[] = "aux";
+    char args[] = "0 1 ";
+    cliAux(cmd, args);
+    vector<string> expected = {
+        "###ERROR IN ", "aux", ": ", "INVALID ARGUMENT COUNT###", "\r\n"
```

Comment:
:-(

Reply:
Could you explain you emoji? Do you want to produce zeros when there is no more data for parsing?

Reply:
Sorry, it's about `flush` calls. This line is terribly stitched together when printing. This is extreme case where parsing output line-by-line will make unittest considerably cleaner.